code/modules/vehicles/train.dm

```diff
@@ -118,17 +118,14 @@
 		return TRUE
 
 /obj/vehicle/train/attack_hand(mob/user)
-	if(user.stat || user.restrained() || !Adjacent(user))
```

Comment:
This can be called from across Z-levels on some objects. Also, `UnarmedAttack()` (which calls `attack_hand()`) doesn't check adjacency/proximity before calling this. The adjacency check is required.